### PR TITLE
Fixed Cargo Bounties

### LINKED ISSUE #955: Cargo bounties dont work
## Description
The trading post isn't owned by the station, meaning the bounties dont count because fuck you. I'm not 100% sure how to fix this yet

### PR DESCRIPTION
**Pull request by blueDev2**
## About the PR

Fixed Cargo Bounties

## Why / Balance

Players tend to want the bounties to work so that they don't have to depend on mail for funds.

## Technical details

Made bounty label components also hold the station where it was printed, so that the main station's bounty DB is used instead of the automated trade station's bounty DB

## Media

TODO

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

Fixes #955

:cl: blueDev2
- fix: Cargo Bounties actually work. Cargo Rejoice!

### CODE REVIEW

File: Content.Server/Cargo/Systems/CargoSystem.Bounty.cs
Comment (Adrian16199) [inline Content.Server/Cargo/Systems/CargoSystem.Bounty.cs]: Im now going to nitpick.
```suggestion
                 entities.Add(child);
```

**Comment by FluffiestFloof** [comment]
Should probably add markers to note what is being changed, and that those are deltav changes.
Also shouldn't this be directly fixed upstream if it's an overall issue?

**Comment by Adrian16199** [comment]
> Should probably add markers to note what is being changed, and that those are deltav changes. Also shouldn't this be directly fixed upstream if it's an overall issue?

To the first thing, yeah, I didnt said much as this is a draft and alot can change so I would wait till its functional.
Regarding the second thing, this is not an upstream issue as debug made automated trading station be different aka it spawning on a different universe compared to the station that you work on in here.

**Comment by deltanedas** [comment]
this still would be nice for upstream so it would support multi station sharing a trade station but using each stations bounties

**Comment by Adrian16199** [comment]
> this still would be nice for upstream so it would support multi station sharing a trade station but using each stations bounties

You are not wrong.

**Comment by blueDev2** [comment]
> Should probably add markers to note what is being changed, and that those are deltav changes. Also shouldn't this be directly fixed upstream if it's an overall issue?

I'm not sure what markers you are talking about.

**Comment by blueDev2** [comment]
Do you think I really should make this an upstream PR?

**Comment by Adrian16199** [comment]
> > Should probably add markers to note what is being changed, and that those are deltav changes. Also shouldn't this be directly fixed upstream if it's an overall issue?
> 
> I'm not sure what markers you are talking about.

The markers are pretty much just notes that indicate that an upstream file was touched that usually goes like: # DELTA V-Fixes Cargo Bountys

**Comment by Adrian16199** [comment]
> Do you think I really should make this an upstream PR?

It would be a good attempt as deltanedas suggested, it would fix it and less upstream file touched that are different from wizden.
I would keep this in draft and just open it if they wont accept it upstream.
I dont think they would deny it as it basicly makes bountys not just go kapoot just because its not on the same galaxy and the code could be used aswell on any other thing for any future ideas.

**Comment by blueDev2** [comment]
> > > Should probably add markers to note what is being changed, and that those are deltav changes. Also shouldn't this be directly fixed upstream if it's an overall issue?
> > 
> > 
> > I'm not sure what markers you are talking about.
> 
> The markers are pretty much just notes that indicate that an upstream file was touched that usually goes like: # DELTA V-Fixes Cargo Bountys

Where am I supposed to mark that down?

**Comment by Adrian16199** [comment]
> Where am I supposed to mark that down?

So basicly

{
  (Wizden code but changed on delta V) # Delta V- Fixes cargo bountys [aka the point of PR, helps with github blame]
  (Wizden code)
  (Wizden code but changed on delta V) # Delta V- Fixes cargo bountys
}
Second example
{
   (Wizden code but changed on delta V) # Delta V- Start of Fixes cargo bountys
   (Wizden code but changed on delta V)
   (Wizden code but changed on delta V) # Delta V- End of Fixes cargo bountys
}

File: Content.Server/Cargo/Components/CargoBountyLabelComponent.cs
Comment (Axionyxx) [inline Content.Server/Cargo/Components/CargoBountyLabelComponent.cs]: Why do you need this imported here?

File: Content.Server/Cargo/Components/CargoBountyLabelComponent.cs
Comment (Axionyxx) [inline Content.Server/Cargo/Components/CargoBountyLabelComponent.cs:22]:
```suggestion
    /// DeltaV - The Station System to check and remove bounties from
```

File: Content.Server/Cargo/Systems/CargoSystem.Bounty.cs
Comment (Axionyxx) [inline Content.Server/Cargo/Systems/CargoSystem.Bounty.cs]: Don't use Serilog

File: Content.Server/Cargo/Systems/CargoSystem.Bounty.cs
Comment (Axionyxx) [inline Content.Server/Cargo/Systems/CargoSystem.Bounty.cs:69]:
```suggestion
        SetupBountyLabel(label, station, bounty.Value); // DeltaV - Track owning station
```

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (Axionyxx) [inline Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs]: You don't need this here

**Review by Axionyxx — CHANGES_REQUESTED** [review]
Here's just a few I had time for right now

**Comment by blueDev2** [comment]
I'm gonna wait until a final decision is made on the upstream PR before adding code markers.
I do appreciate you giving me some examples though.

**Comment by stellar-novas** [comment]
> Do you think I really should make this an upstream PR?

Yes, this is a flaw with the bounty system that would also affect wizden, and their downstreams, if they put the station on a different map.

If they don't accept it for whatever reason, which I doubt, can just reopen this pr.

**Comment by blueDev2** [comment]
Upstream PR merged: https://github.com/space-wizards/space-station-14/pull/26469
I will close this PR and make a new PR by pulling the 3 applicable upstream files.

**Pull request closed without merging**
